### PR TITLE
Generate typescript api dancelight

### PR DESCRIPTION
**Pull request by girazoki**
Adds a few things:

- the possibility to add `--dev-service` to the tanssi-relay binary so that we can launch a custom config in dev-mode. This is similar to what we have in tanssi-node

- the typescript generation for dancelight

### CODE REVIEW

File: typescript-api/scripts/runtime-upgrade.ts [hunk 1]
```diff
@@ -26,18 +26,34 @@ async function main() {
...
+                "--no-prometheus",
+                "--alice",
+                "--tmp",
+                `--chain=${CHAIN}-local`,
+                "--dev-service",
+                "--wasm-execution=interpreted-i-know-what-i-do",
+                "--rpc-port=9933",
+                "--unsafe-force-node-key-generation",
+            ]);
+        }
```
Comment (ekenigs) [inline typescript-api/scripts/runtime-upgrade.ts:56]:
```suggestion
    const isStarlightChain = CHAIN.includes("light");

    nodeProcess = spawn(`../target/release/tanssi-${isStarlightChain ? 'relay' : 'node'}`, [
      "--no-hardware-benchmarks",
      "--no-telemetry",
      "--no-prometheus",
      "--alice",
      "--tmp",
      `--chain=${CHAIN}-local`,
      "--dev-service",
      "--wasm-execution=interpreted-i-know-what-i-do",
      "--rpc-port=9933",
      "--unsafe-force-node-key-generation",
    ]);
```

File: typescript-api/package.json
Comment (ekenigs) [inline typescript-api/package.json:1]: You need to add `dancelight` to `exports` below `flashbox` export.

Comment (girazoki) [reply typescript-api/package.json:1]: fixed on [f43e732](https://github.com/moondance-labs/tanssi/pull/711/commits/f43e73281fdce52540f62a654f1b22bb1e1a0440)

**Review by ekenigs — APPROVED** [review]
LGTM

**Pull request merged**